### PR TITLE
Feature/ecp orbital

### PR DESCRIPTION
**Pull request by ajawadmirza**
ECP Chase Orbital-Electronic Check Integration:

1. ECP Standard Processing
2. ECP Authorization Methods
3. ECPSameDay Index Processing

### CODE REVIEW

**Comment by ajawadmirza** [comment]
**ECP Chase Orbital-Electronic Check Integration:**

- ECP Standard Processing
- ECP Authorization Methods
- ECPSameDay Index Processing

File: test/remote/gateways/remote_orbital_test.rb
Comment (jessiagee) [inline test/remote/gateways/remote_orbital_test.rb:498]: Do you need all of this blank space after APPROVAL?

File: test/remote/gateways/remote_orbital_test.rb
Comment (jessiagee) [inline test/remote/gateways/remote_orbital_test.rb:615]: Should this be 123;123 instead of 3F3F3F ?

File: test/unit/gateways/orbital_test.rb
Comment (jessiagee) [inline test/unit/gateways/orbital_test.rb:141]: Do we need all of this white space behind APPROVAL?

File: test/unit/gateways/orbital_test.rb
Comment (jessiagee) [inline test/unit/gateways/orbital_test.rb:946]: Is '1;2' the correct param to pass? I don't see the difference between that and the failure below.

**Review by jessiagee — CHANGES_REQUESTED** [review]

File: lib/active_merchant/billing/gateways/orbital.rb
Comment (ajawadmirza) [reply lib/active_merchant/billing/gateways/orbital.rb:507]: If we do this in the current code: 
`
        if payment_source.instance_of?(ActiveMerchant::Billing::Check)
          add_echeck(xml, payment_source, options)
        elsif payment_source.instance_of?(ActiveMerchant::Billing::CreditCard) or payment_source.instance_of?(ActiveMerchant::Billing::NetworkTokenizationCreditCard)
          add_creditcard(xml, payment_source, options[:currency])
        end
`

then in that case, Orbital's previously existing two unit test cases are failed as in the previous implementation CurrencyCode needs to be added even when no payment source is passed (please refer to image below):

image

Now, we can have three possibilities here:
1. Keep it as it is so that everything works fine as in the case of no credit card, some required attributes will be added
2. Make changes in previous Orbital code and debug so that it should work even when no payment source is added
3. Make changes in the tests of the previous orbital's implementation

Please suggest on this?

File: test/remote/gateways/remote_orbital_test.rb
Comment (ajawadmirza) [reply test/remote/gateways/remote_orbital_test.rb:498]: Blank spaces are already produced in our response, that's why it was put here to match it exactly. However, we can the condition so that it checks whether APPROVAL is contained in the response.message.

File: test/remote/gateways/remote_orbital_test.rb
Comment (ajawadmirza) [reply test/remote/gateways/remote_orbital_test.rb:615]: Whatever value we provide in the authorization string, server produces response that has 3F3F3F and please note that 123 doesn't mean something special but it just an invalid number passed to check the failure case.

File: test/unit/gateways/orbital_test.rb
Comment (ajawadmirza) [reply test/unit/gateways/orbital_test.rb:141]: Already answered this before.

File: test/unit/gateways/orbital_test.rb
Comment (ajawadmirza) [reply test/unit/gateways/orbital_test.rb:946]: @jessiagee , please note that this is a unit test, where we produce static response to test the correct rendering of the response. Here the response is already given in the **successful_refund_with_echeck_response**. So, the values are just passed so that the method can be called correctly and produce our static response.

Previous one was an actual remote test case where we hit the server and get the response, so in that case we intentionally gave incorrect value to test the failure (negative) case.

File: lib/active_merchant/billing/gateways/orbital.rb
Comment (jessiagee) [reply lib/active_merchant/billing/gateways/orbital.rb:517]: If there's existing code that this check would break than let's avoid adding that in. I don't want to mess with any legacy code that could potentially cause a lot more issues to existing customers. Thanks!

File: test/remote/gateways/remote_orbital_test.rb
Comment (jessiagee) [reply test/remote/gateways/remote_orbital_test.rb:615]: Okay thanks for the information!

File: test/unit/gateways/orbital_test.rb
Comment (jessiagee) [reply test/unit/gateways/orbital_test.rb:946]: Thanks for the clarification!

**Review by jessiagee — APPROVED** [review]

**Pull request closed without merging**
